controllers/cruisecontroloperation_controller.go

```diff
@@ -0,0 +1,473 @@
+			operation.Status.FailedTasks = append(operation.Status.FailedTasks[1:], *task)
+		} else {
+			operation.Status.FailedTasks = append(operation.Status.FailedTasks, *task)
+		}
+		operation.Status.NumberOfRetries += 1
+		task.SetDefaults()
+	}
+
+	if isAfterExecution {
+		if task.Started == nil {
+			startTime, err := time.Parse(time.RFC1123, res.StartedAt)
+			if err != nil {
+				return errors.WrapIff(err, "could not parse user task start time from Cruise Control API")
+			}
+			task.Started = &v1.Time{Time: startTime}
+		}
+		task.ID = res.TaskID
+		task.Summary = parseSummary(res.Result)
+		task.ErrorMessage = res.Err
+		task.HTTPRequest = res.RequestURL
+		task.HTTPResponseCode = &res.ResponseStatusCode
+	}
+
+	task.State = res.State
+
+	return nil
+}
+
+// updateActiveTasks updates the state of the CruiseControlOperation from the CruiseControlTasksAndStates instance by getting their
+// status from Cruise Control.
+func (r *CruiseControlOperationReconciler) updateCurrentTasks(ctx context.Context, ccOperations []*banzaiv1alpha1.CruiseControlOperation) error {
+	log := logr.FromContextOrDiscard(ctx)
+
+	userTaskIDs := make([]string, 0, len(ccOperations))
+	var ccOperationsCopy []*banzaiv1alpha1.CruiseControlOperation
+	for _, ccOperation := range ccOperations {
+		if ccOperation.GetCurrentTaskID() != "" {
+			userTaskIDs = append(userTaskIDs, ccOperation.GetCurrentTaskID())
+		}
+		copy := ccOperation.DeepCopy()
+		ccOperationsCopy = append(ccOperationsCopy, copy)
+	}
+
+	tasks, err := r.Scaler.GetUserTasks(userTaskIDs...)
+	if err != nil {
+		return errors.WrapIff(err, "could not get user tasks from Cruise Control API")
+	}
+
+	taskResultsByID := make(map[string]*scale.Result, len(tasks))
+	for _, task := range tasks {
+		taskResultsByID[task.TaskID] = task
+	}
+	for i := range ccOperations {
+		ccOperation := ccOperations[i]
+		if ccOperation.GetCurrentTaskID() != "" && !ccOperation.IsDone() {
+			if err := updateResult(log, taskResultsByID[ccOperation.GetCurrentTaskID()], ccOperation, false); err != nil {
+				return errors.WrapWithDetails(err, "could not set Cruise Control user task result to CruiseControlOperation CurrentTask", "name", ccOperations[i].GetName(), "namespace", ccOperations[i].GetNamespace())
+			}
+		}
+	}
+
+	for i := range ccOperations {
+		if !reflect.DeepEqual(ccOperations[i].Status, ccOperationsCopy[i].Status) {
+			if err := r.Status().Update(ctx, ccOperations[i]); err != nil {
+				return errors.WrapIfWithDetails(err, "could not update CruiseControlOperation status", "name", ccOperations[i].GetName(), "namespace", ccOperations[i].GetNamespace())
+			}
+		}
+	}
+	return nil
+}
+
+func isWaitingForFinalize(ccOperation *banzaiv1alpha1.CruiseControlOperation) bool {
+	if ccOperation.IsCurrentTaskRunning() && !ccOperation.ObjectMeta.DeletionTimestamp.IsZero() && controllerutil.ContainsFinalizer(ccOperation, cruiseControlOperationFinalizer) {
+		return true
+	}
+	return false
```

Comment:
same logic but shorter
```suggestion
	return ccOperation.IsCurrentTaskRunning() && !ccOperation.ObjectMeta.DeletionTimestamp.IsZero() && controllerutil.ContainsFinalizer(ccOperation, cruiseControlOperationFinalizer)
```